bridge_common/validateapi.py

```diff
@@ -0,0 +1,191 @@
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Refer to the README and COPYING files for full details of the license
+
+# Usage: opr = SdsOperations()
+#        opr.validateApi(api-job)
+# This api can be used to validate the api-job against the
+# perdefined sds_operation_schema in the yaml file.
+
+
+import os
+import six
+import yaml
+
+PRIMITIVE_TYPES = {'Boolean': lambda value: isinstance(value, bool),
+                   'Float': lambda value: isinstance(value, float),
+                   'Integer': lambda value: isinstance(value, int),
+                   'Long': lambda value: isinstance(value, (
+                       six.integer_types, float)),
+                   'String': lambda value: isinstance(value, six.string_types),
+                   'Uint': lambda value: isinstance(value, int) and value >= 0,
+                   'list': lambda value: isinstance(value, list)}
+
+
+class SchemaNotFound(Exception):
+    pass
+
+
+def locateSchema(schemaName='operations'):
+    """
+    Locate the bridge operation schema file whether we are running
+    from within the source dir or from installed location
+    """
+    localpath = os.path.dirname(__file__)
+    installedpath = "/usr/local/share/tendrl/"
+    for directory in (localpath, installedpath):
+        path = os.path.join(directory, "sds_state_" + schemaName + '.yaml')
+        # we use source tree and deployment directory
+        # so we need to check whether file exists
+        if os.path.exists(path):
+            return path
+
+        raise SchemaNotFound("Unable to find API schema file in %s or %s",
+                             localpath, installedpath)
+
+
+def loadSchema(schemaName):
+    schemaFile = locateSchema(schemaName)
+    try:
+        # Check api operation schema file exists
+        if not os.path.isfile(schemaFile):
+            return (False,
+                    "Error: Schema file '" + schemaFile + "' does not exists.")
+
+        # load api operation schema file and return
+        try:
+            code = open(schemaFile)
+        except IOError, exc:
+            return (False,
+                    "Error loading schema file '" + schemaFile + "': " + exc)
+
+        # Parse schema file
+        try:
+            config = yaml.load(code)
+        except yaml.YAMLError, exc:
+            error_pos = ""
+            if hasattr(exc, 'problem_mark'):
+                error_pos = " at position: (%s:%s)" % (
+                    exc.problem_mark.line+1,
+                    exc.problem_mark.column+1)
+            msg = "Error loading schema file '" + schemaFile + "'" + error_pos \
+                + ": content format error: Failed to parse yaml format"
+            return False, msg
+
+    except Exception, e:
+        return (False, "Error loading api operation schema file '%s': %s" % (
+            schemaFile, str(e)))
+
+    return (True, config)
+
+
+def validateApiWithYamlConf(apiJob, yamlConf):
```

Comment:
yamlConf should be class level attribute (say class name is ApiJobValidator) and need not be passed every time to function here.
The caller would execute something like

---------------
from tendrl.bridge_common.validateapi import ApiJobValidator

validator = ApiJobValidator('<Full path of the yaml file>')
ret = validator.Validate(apijob)
if not ret:
    Failed